hazelcast/src/main/java/com/hazelcast/internal/adapter/DataStructureAdapter.java

```diff
@@ -56,4 +56,43 @@
     LocalMapStats getLocalMapStats();
 
     boolean containsKey(K key);
+
+    /**
+     * Contains all methods of {@link DataStructureAdapter}.
+     */
+    enum DataStructureMethods implements DataStructureAdapterMethod {
```

Comment:
Yeah, I tried some reflection magic as well, but had ran into the same issue. I'll try to come up with a test which doesn't need too much maintenance on it's own.